### PR TITLE
107857 swaggerdoc

### PR DESCRIPTION
**Pull request by janid1967**
[AB#107857](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/107857)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public static IEnumerable<PropertyInfo> CollectPropertiesWithAttribute(this Type type, Type attributeType)
+    {
+        var props = GetWritableProperties(type);
+        return CollectPropertiesWithAttribute(props, attributeType);
+    }
+
+    public static IEnumerable<PropertyInfo> GetWritableProperties(this Type type)
+        => type.GetProperties(BindingFlags.Instance | BindingFlags.Public).Where(p => p.CanWrite);
+
+    public static IEnumerable<PropertyInfo> CollectPropertiesWithStringLengthAttribute(this Type type)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs:20]: Should we rename to CollectWritablePropertiesWithStringLengthAttribute to make it crystal clear what is happening under the covers?

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace Equinor.ProCoSys.Completion.WebApi.Swagger;
+
+public class SwaggerDocHelper
+{
+    public static void FillPatchDocumentWithSampleData<T>(JsonPatchDocument<T> patchDocument) where T : class
+    {
+        var type = typeof(T);
+        var props = type.GetProperties(BindingFlags.Instance | BindingFlags.Public)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs:14]: Can use TypeExtension GetWritableProperties method instead?

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            .Where(p => p.CanWrite);
+        //var requiredProps = props.Where(prop => prop.IsDefined(typeof(RequiredAttribute), false))
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs:16]: Commented out code...

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+public class SwaggerPatchDocumentation : IOperationFilter
+{
+    private readonly string _generalPatchInfo =
+        "Patching can be done on zero properties, one property only, all properties in once, or any combination of properties.<br>" +
+        "If a required property is patched, a (new) value must be given.<br>" +
+        "If an optional property is patched, both a (new) value or null can be given. Empty string can also be used for string properties.<br>" + 
+        "<br>" +
+        "Only replace operation are supported (\"op\": \"replace\").<br>";
+        
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs:21]: Is it possible to write unit tests for this metod?

File: src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using System;
+using System.Linq;
+using System.ComponentModel.DataAnnotations;
+
+namespace Equinor.ProCoSys.Completion.WebApi.Misc;
+
+public static class TypeExtension
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs:9]: Possible to unit test?

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.AspNetCore.JsonPatch;
+using System.Reflection;
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace Equinor.ProCoSys.Completion.WebApi.Swagger;
+
+public class SwaggerDocHelper
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs:9]: Possible to unit test?

Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerDocHelper.cs:9]: Will fix

File: src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs (hunk 5 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.WebApi/Misc/TypeExtension.cs:9]: Will fix

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs (hunk 4 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs:21]: Unsure, is it worth it? Not for end users

**Review by jsoi-eq — APPROVED** [review]
Approving, pending adding some unit tests.

File: src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs (hunk 4 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.WebApi/Swagger/SwaggerPatchDocumentation.cs:21]: No tests written

**Pull request merged**